### PR TITLE
Multiple views

### PR DESCRIPTION
**Pull request by rupakbajgain**

### CODE REVIEW

**Comment by rupakbajgain** [comment]
Now there should be common snap managers.... And others...

**Comment by rupakbajgain** [comment]
If we zoom out too much the background automatically changes to white.

File: lckernel/cad/storage/documentimpl.cpp [hunk 1]
```diff
@@ -91,6 +94,13 @@ void DocumentImpl::addDocumentMetaType(const lc::meta::DocumentMetaType_CSPtr& d
         addLayerEvent()(event);
     }
 
+	auto viewport = std::dynamic_pointer_cast<const meta::Viewport>(dmt);
+    if (layer != nullptr) {
```
Comment (feragon) [inline lckernel/cad/storage/documentimpl.cpp]: I think you meant `viewport != nullptr` here :)

**Comment by feragon** [comment]
The storage looks fine.

Now you have to connect the kernel events (AddViewportEvent) to your LCADProxyViewer to create the tabs automatically. If you need an example, look at the Layers widget: https://github.com/LibreCAD/LibreCAD_3/blob/master/lcUI/widgets/layers.cpp#L43-L45

**Comment by rupakbajgain** [comment]
Now even the entities are seperated they have drag point in another viewport.

**Comment by rupakbajgain** [comment]
Loading file is not loading in model space.

**Pull request merged**